crates/python/src/executable.rs

```diff
@@ -0,0 +1,154 @@
+use std::sync::Arc;
+
+use pyo3::{
+    exceptions::PyRuntimeError, pyclass, pymethods, types::PyDict, FromPyObject, Py, PyAny,
+    PyResult, Python,
+};
+use qcs::{Error, Executable, JobHandle, Qpu, Qvm, Service};
+use rigetti_pyo3::{
+    impl_as_mut_for_wrapper, py_wrap_error, py_wrap_simple_enum, py_wrap_type, wrap_error,
+    PyWrapper, ToPython, ToPythonError,
+};
+use tokio::sync::Mutex;
+
+use crate::qpu::quilc::PyCompilerOpts;
+
+wrap_error!(ExecutionError(Error));
+
+py_wrap_error!(
+    executable,
+    ExecutionError,
+    QcsExecutionError,
+    PyRuntimeError
+);
+
+// Because Python is garbage-collected, no lifetimes can be guaranteed except `'static`.
+//
+// `Arc<Mutex<>>` to work around https://github.com/awestlake87/pyo3-asyncio/issues/50
+py_wrap_type! {
+    PyExecutable(Arc<Mutex<Executable<'static, 'static>>>) as "Executable";
+}
```

Comment:
`pyo3_asyncio` requires that all futures be `'static`, which does not work when the future has a reference to `&self`. The workaround for this is to wrap the inner data in an `Arc<Mutex<_>>`, which effectively gives the inner data the `'static` lifetime.